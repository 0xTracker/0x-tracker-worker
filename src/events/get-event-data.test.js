const { UnsupportedAssetError } = require('../errors');
const getEventData = require('./get-event-data');
const V2_EVENT = require('../fixtures/events/v2');
const V3_EVENT = require('../fixtures/events/v3');

const eventWithArgs = (event, args) => ({
  ...event,
  data: {
    ...event.data,
    args: {
      ...event.data.args,
      ...args,
    },
  },
});

describe('getEventData', () => {
  it('should get data for V2 event', () => {
    const normalized = getEventData(V2_EVENT);
    expect(normalized).toMatchSnapshot();
  });

  it('should get data for V3 event', () => {
    const normalized = getEventData(V3_EVENT);
    expect(normalized).toMatchSnapshot();
  });

  it('should throw UnsupportedAssetError when maker asset data is invalid', () => {
    const event = eventWithArgs(V2_EVENT, { makerAssetData: 'invalid' });

    expect(() => {
      getEventData(event);
    }).toThrow(UnsupportedAssetError);
  });

  it('should throw UnsupportedAssetError when taker asset data is invalid', () => {
    const event = eventWithArgs(V2_EVENT, { takerAssetData: 'invalid' });

    expect(() => {
      getEventData(event);
    }).toThrow(UnsupportedAssetError);
  });
});
