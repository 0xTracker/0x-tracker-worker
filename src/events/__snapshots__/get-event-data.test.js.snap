// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeFillArgs should normalize args for V1 event 1`] = `
Object {
  "assets": Array [
    Object {
      "actor": 0,
      "amount": 1292498732163907300,
      "tokenAddress": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
      "tokenResolved": false,
      "tokenType": 0,
    },
    Object {
      "actor": 1,
      "amount": 4.2620995342647e+23,
      "tokenAddress": "0x58a4884182d9e835597f405e5f258290e46ae7c2",
      "tokenResolved": false,
      "tokenType": 0,
    },
  ],
  "blockHash": "0xd383bf2c68bfd74d777857fc066390d1f18934cd8ad57a967b5a8049ff8bd479",
  "blockNumber": 6062314,
  "feeRecipient": "0xe269e891a2ec8585a378882ffa531141205e92e9",
  "fees": Array [
    Object {
      "amount": Object {
        "token": "1.5",
      },
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
      "tokenType": 0,
      "traderType": 0,
    },
    Object {
      "amount": Object {
        "token": "1.3",
      },
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
      "tokenType": 0,
      "traderType": 1,
    },
  ],
  "logIndex": 101,
  "maker": "0x0e7c68992fe035f4ed9607866483cf6ae7db6b71",
  "orderHash": "0x9af44f4536831684cacda6dbba6af9974f0b9e8b2fc271f4fa8ff18c6fd51e12",
  "taker": "0xe269e891a2ec8585a378882ffa531141205e92e9",
  "transactionHash": "0x00cfc187cce6c5f537f84621b6fce4ac828848f2b088b16f0deeb4bde2586637",
}
`;

exports[`normalizeFillArgs should normalize args for V2 event with recognized assets 1`] = `
Object {
  "assets": Array [
    Object {
      "actor": 0,
      "amount": 1,
      "tokenAddress": "0x06012c8cf97bead5deae237070f9587f8e7a266d",
      "tokenId": 862718,
      "tokenResolved": false,
      "tokenType": 1,
    },
    Object {
      "actor": 1,
      "amount": 100000000000000,
      "tokenAddress": "0xe41d2489571d322189246dafa5ebde1f4699f498",
      "tokenResolved": false,
      "tokenType": 0,
    },
  ],
  "blockHash": "0x592bdb8653f20b291b3cf927314344f299c6e37a3a2887a558b29584d60730d6",
  "blockNumber": 6286241,
  "feeRecipient": "0x0000000000000000000000000000000000000000",
  "fees": Array [],
  "logIndex": 42,
  "maker": "0x5b50fb8281cb679a3102d3de22dd05a1ad895e8f",
  "orderHash": "0xb0d28d8c36f40322408b3ff4ac41d2bf6ac6c834633763d3b42831d50e30b455",
  "senderAddress": "0xf3f694cbe3a1042a886ce923fab00ca649e6a89b",
  "taker": "0xf3f694cbe3a1042a886ce923fab00ca649e6a89b",
  "transactionHash": "0x28ffb48f354997d384eee49d326c13a10c4584ca3bced4632053b201d3a0cbbc",
}
`;
